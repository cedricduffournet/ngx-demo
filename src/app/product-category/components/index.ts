export * from './product-category-add/product-category-add.component';
export * from './product-category-update/product-category-update.component';
export * from './product-category-delete/product-category-delete.component';
export * from './product-category-form/product-category-form.component';
export * from './product-category-item/product-category-item.component';
export * from './product-category-items/product-category-items.component';
