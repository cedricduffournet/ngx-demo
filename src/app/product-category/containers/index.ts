export * from './product-category-add-modal/product-category-add-modal.component';
export * from './product-category-update-modal/product-category-update-modal.component';
export * from './product-category-delete-modal/product-category-delete-modal.component';
export * from './product-category-list-view/product-category-list-view.component';
