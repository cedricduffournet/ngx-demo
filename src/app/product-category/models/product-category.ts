import { schema } from 'normalizr';

export interface ProductCategory {
  id: number;
  name: string;
}

export const productCategorySchema = new schema.Entity('productCategories');
