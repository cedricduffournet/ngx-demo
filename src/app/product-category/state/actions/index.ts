import * as ProductCategoryUpdateModalActions from './product-category-update-modal.actions';
import * as ProductCategoryAddModalActions from './product-category-add-modal.actions';
import * as ProductCategoryDeleteModalActions from './product-category-delete-modal.actions';
import * as ProductCategoryListViewActions from './product-category-list-view.actions';
import * as ProductCategoryApiActions from './product-category-api.actions';

export {
  ProductCategoryListViewActions,
  ProductCategoryApiActions,
  ProductCategoryUpdateModalActions,
  ProductCategoryAddModalActions,
  ProductCategoryDeleteModalActions
};
