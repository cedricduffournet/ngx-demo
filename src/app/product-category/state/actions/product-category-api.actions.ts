import { props, createAction } from '@ngrx/store';

import { NormalizedData } from '@app/shared/models/normalized.model';

export const loadProductCategorySuccess = createAction(
  '[ProductCategory API] Load SUCCESS',
  props<{ productCategories: NormalizedData }>()
);

export const loadProductCategoryFailure = createAction(
  '[ProductCategory API] Load FAILURE',
  props<{ error: any }>()
);

export const addProductCategorySuccess = createAction(
  '[ProductCategory API] Add SUCCESS',
  props<{ productCategory: NormalizedData }>()
);

export const addProductCategoryFailure = createAction(
  '[ProductCategory API] Add FAILURE',
  props<{ error: any }>()
);

export const updateProductCategorySuccess = createAction(
  '[ProductCategory API] Update  SUCCESS',
  props<{ productCategory: NormalizedData }>()
);

export const updateProductCategoryFailure = createAction(
  '[ProductCategory API] Update FAILURE',
  props<{ error: any }>()
);

export const deleteProductCategorySuccess = createAction(
  '[ProductCategory API] Delete SUCCESS',
  props<{ id: number }>()
);

export const deleteProductCategoryFailure = createAction(
  '[ProductCategory API] Delete FAILURE',
  props<{ error: any }>()
);
