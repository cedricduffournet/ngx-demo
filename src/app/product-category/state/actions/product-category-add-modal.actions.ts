import { props, createAction } from '@ngrx/store';

import { ProductCategory } from '@app/product-category/models/product-category';

export const addProductCategory = createAction(
  '[ProductCategory add  modal] Add',
  props<{ productCategory: ProductCategory }>()
);
