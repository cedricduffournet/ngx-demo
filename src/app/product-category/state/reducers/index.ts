import {
  createSelector,
  createFeatureSelector,
  Action,
  combineReducers
} from '@ngrx/store';

import * as fromRoot from '@app/core/state/reducers';
import * as fromProductCategoryCollection from '@app/product-category/state/reducers/product-category-collection.reducer';
import * as fromProductCategoryEntities from '@app/product-category/state/reducers/product-category-entities.reducer';
import { ProductCategory } from '@app/product-category/models/product-category';

export interface ProductCategoriesState {
  collection: fromProductCategoryCollection.State;
  productCategories: fromProductCategoryEntities.State;
}

export interface State extends fromRoot.State {
  productCategories: ProductCategoriesState;
}

export function reducers(state: ProductCategoriesState | undefined, action: Action) {
  return combineReducers({
    collection: fromProductCategoryCollection.reducer,
    productCategories: fromProductCategoryEntities.reducer
  })(state, action);
}

export const selectProductCategoriesState = createFeatureSelector<
  State,
  ProductCategoriesState
>('productCategories');

export const getProductCategoryEntitiesState = createSelector(
  selectProductCategoriesState,
  state => state.productCategories
);

export const getProductCategoryEntities = createSelector(
  getProductCategoryEntitiesState,
  fromProductCategoryEntities.getEntities
);

export const getProductCategoryEntitiesUpdating = createSelector(
  getProductCategoryEntitiesState,
  fromProductCategoryEntities.getUpdating
);

export const getProductCategoryEntitiesUpdated = createSelector(
  getProductCategoryEntitiesState,
  fromProductCategoryEntities.getUpdated
);

export const getProductCategoryCollectionState = createSelector(
  selectProductCategoriesState,
  state => state.collection
);

export const getProductCategoryIds = createSelector(
  getProductCategoryCollectionState,
  fromProductCategoryCollection.getIds
);

export const getProductCategoryCollectionLoading = createSelector(
  getProductCategoryCollectionState,
  fromProductCategoryCollection.getLoading
);

export const getProductCategoryCollectionLoaded = createSelector(
  getProductCategoryCollectionState,
  fromProductCategoryCollection.getLoaded
);

export const getProductCategoryCollectionAdding = createSelector(
  getProductCategoryCollectionState,
  fromProductCategoryCollection.getAdding
);

export const getProductCategoryCollectionAdded = createSelector(
  getProductCategoryCollectionState,
  fromProductCategoryCollection.getAdded
);

export const getProductCategoryCollectionDeleting = createSelector(
  getProductCategoryCollectionState,
  fromProductCategoryCollection.getDeleting
);

export const getProductCategoryCollectionDeleted = createSelector(
  getProductCategoryCollectionState,
  fromProductCategoryCollection.getDeleted
);

export const getProductCategories = createSelector(
  getProductCategoryEntities,
  getProductCategoryIds,
  (entities, ids) => {
    return ids.map(id => entities[id]);
  }
);

export const getSelectedProductCategoryId = createSelector(
  getProductCategoryEntitiesState,
  fromProductCategoryEntities.getSelectedId
);

export const getSelectedProductCategory = createSelector(
  getProductCategoryEntities,
  getSelectedProductCategoryId,
  (productCategoryEntities, productCategoryId): ProductCategory => {
    return productCategoryEntities[productCategoryId];
  }
);
