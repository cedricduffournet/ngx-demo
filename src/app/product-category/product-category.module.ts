import { NgModule, ComponentFactoryResolver } from '@angular/core';

import { RouterModule } from '@angular/router';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule } from '@ngrx/store';

import { SharedModule } from '@app/shared/shared.module';
import { ProductCategoryEffects } from '@app/product-category/state/effects';
import {
  ProductCategoryAddComponent,
  ProductCategoryUpdateComponent,
  ProductCategoryDeleteComponent,
  ProductCategoryFormComponent,
  ProductCategoryItemComponent,
  ProductCategoryItemsComponent
} from '@app/product-category/components';
import {
  ProductCategoryAddModalComponent,
  ProductCategoryUpdateModalComponent,
  ProductCategoryDeleteModalComponent,
  ProductCategoryListViewComponent
} from '@app/product-category/containers';
import { reducers } from '@app/product-category/state/reducers';
import { CoalescingComponentFactoryResolver } from '@app/coalescing-component-factory-resolver.service';

@NgModule({
  imports: [
    RouterModule,
    SharedModule,
    EffectsModule.forFeature([ProductCategoryEffects]),
    StoreModule.forFeature('productCategories', reducers)
  ],
  declarations: [
    ProductCategoryListViewComponent,
    ProductCategoryAddComponent,
    ProductCategoryUpdateComponent,
    ProductCategoryDeleteComponent,
    ProductCategoryFormComponent,
    ProductCategoryItemComponent,
    ProductCategoryItemsComponent,
    ProductCategoryAddModalComponent,
    ProductCategoryUpdateModalComponent,
    ProductCategoryDeleteModalComponent
  ],
  entryComponents: [
    ProductCategoryUpdateModalComponent,
    ProductCategoryAddModalComponent,
    ProductCategoryDeleteModalComponent
  ],
  exports: [ProductCategoryListViewComponent]
})
export class ProductCategoryModule {
  // see https://github.com/angular/angular/issues/14324
  constructor(
    coalescingResolver: CoalescingComponentFactoryResolver,
    localResolver: ComponentFactoryResolver
  ) {
    coalescingResolver.registerResolver(localResolver);
  }
}
