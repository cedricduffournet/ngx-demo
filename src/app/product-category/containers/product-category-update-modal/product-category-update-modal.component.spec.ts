import { TestBed, ComponentFixture } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';

import { BsModalRef } from 'ngx-bootstrap/modal';
import { Store, MemoizedSelector } from '@ngrx/store';
import { MockStore, provideMockStore } from '@ngrx/store/testing';
import { TranslateModule } from '@ngx-translate/core';

import { FormModule } from '@app/shared/form';
import { ValidationActionModule } from '@app/shared/validation-action';
import { ModalWrapperModule } from '@app/shared/modal';

import {
  ProductCategoryUpdateComponent,
  ProductCategoryFormComponent
} from '@app/product-category/components';
import { ProductCategoryUpdateModalComponent } from '@app/product-category/containers';
import * as fromCivilities from '@app/product-category/state/reducers';

import { ProductCategoryUpdateModalActions } from '@app/product-category/state/actions';
import { ProductCategory } from '@app/product-category/models/product-category';

describe('UpdateProductCategoryModalComponent', () => {
  let fixture: ComponentFixture<ProductCategoryUpdateModalComponent>;
  let component: ProductCategoryUpdateModalComponent;
  let store: MockStore<fromCivilities.State>;
  let updated: MemoizedSelector<fromCivilities.State, boolean>;
  let productCategory: MemoizedSelector<fromCivilities.State, ProductCategory>;
  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [
        ProductCategoryUpdateComponent,
        ProductCategoryUpdateModalComponent,
        ProductCategoryFormComponent
      ],
      imports: [
        TranslateModule.forRoot(),
        FormModule,
        ValidationActionModule,
        ModalWrapperModule,
        ReactiveFormsModule
      ],
      providers: [provideMockStore(), BsModalRef]
    });

    fixture = TestBed.createComponent(ProductCategoryUpdateModalComponent);
    component = fixture.componentInstance;
    store = TestBed.get(Store);
    updated = store.overrideSelector(
      fromCivilities.getProductCategoryEntitiesUpdated,
      false
    );
    productCategory = store.overrideSelector(fromCivilities.getSelectedProductCategory, {
      id: 1,
      name: 'name'
    });
    spyOn(store, 'dispatch');
  });

  it('should be created', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should dispatch a deleteProductCategory event on submit', () => {
    const data = {
      id: 1,
      productCategory: {
        name: 'name'
      } as ProductCategory
    };
    const action = ProductCategoryUpdateModalActions.updateProductCategory({ data });
    fixture.detectChanges();
    component.onUpdate(data);

    expect(store.dispatch).toHaveBeenCalledWith(action);
  });

  it('should close modal after productCategory added', () => {
    spyOn(component.bsModalRef, 'hide');
    fixture.detectChanges();
    updated.setResult(true);
    store.setState({} as any);
    expect(component.bsModalRef.hide).toHaveBeenCalled();
  });

  it('should close modal on cancel', () => {
    spyOn(component.bsModalRef, 'hide');
    component.onCancel();
    expect(component.bsModalRef.hide).toHaveBeenCalled();
  });
});
