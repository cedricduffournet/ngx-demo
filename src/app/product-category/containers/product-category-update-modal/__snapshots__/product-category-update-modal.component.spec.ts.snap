// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateProductCategoryModalComponent should be created 1`] = `
<app-product-category-update-modal
  bsModalRef={[Function BsModalRef]}
  facade={[Function Object]}
  selectedProductCategory$={[Function Observable]}
  subscription={[Function Subscriber]}
  updated$={[Function Observable]}
  updating$="undefined"
>
  <app-modal-wrapper
    ng-reflect-title="EDIT_PRODUCT_CATEGORY_TITLE"
  >
    <div
      class="modal-content bg-light"
    >
      
      <div
        class="modal-header"
      >
        <h4
          class="modal-title"
        >
          EDIT_PRODUCT_CATEGORY_TITLE
        </h4>
      </div>
      <div
        class="modal-body p-0"
      >
        <app-product-category-update
          ng-reflect-product-category="[object Object]"
        >
          <app-product-category-form
            ng-reflect-product-category="[object Object]"
          >
            <form
              class="ng-untouched ng-pristine ng-valid"
              ng-reflect-form="[object Object]"
              novalidate=""
            >
              <section
                class="bg-white"
              >
                <div
                  class="p-3"
                >
                  <app-text-input
                    class="ng-untouched ng-pristine ng-valid"
                    formcontrolname="name"
                    ng-reflect-label="NAME"
                    ng-reflect-name="name"
                    ng-reflect-submitted="false"
                  >
                    <div
                      class="form-group "
                    >
                      <label
                        class="control-label"
                      >
                        NAME
                      </label>
                      <input
                        class="form-control"
                        placeholder=""
                        type="text"
                      />
                      
                      
                      
                      
                      
                      
                    </div>
                  </app-text-input>
                </div>
              </section>
              <app-wrapper-validation-action>
                <div
                  class="d-flex justify-content-end align-items-center p-3"
                >
                  <app-validation-button
                    ng-reflect-label-cancel="CANCEL"
                    ng-reflect-label-confirm="SAVE"
                  >
                    <div
                      class="button-validation-action text-center d-flex justify-content-end align-items-center p-3-group"
                      ng-reflect-klass="button-validation-action text-"
                      ng-reflect-ng-class=""
                    >
                      
                      <button
                        class="btn btn-cancel btn-lg btn-light"
                        ng-reflect-klass="btn btn-cancel btn-lg"
                        ng-reflect-ng-class="btn-light"
                        type="button"
                      >
                         CANCEL 
                      </button>
                      
                      <button
                        class="btn btn-confirm btn-lg btn-primary"
                        ng-reflect-klass="btn btn-confirm btn-lg"
                        ng-reflect-ng-class="btn-primary"
                        type="submit"
                      >
                        
                        
                        
                        <span>
                          SAVE
                        </span>
                      </button>
                    </div>
                  </app-validation-button>
                </div>
              </app-wrapper-validation-action>
            </form>
          </app-product-category-form>
        </app-product-category-update>
      </div>
    </div>
  </app-modal-wrapper>
</app-product-category-update-modal>
`;
